script.js

```diff
@@ -0,0 +1,356 @@
+/* global SpatialNavigator */
+
+window.SmartScrollbox = (function(exports) {
+  'use strict';
+
+  var proto = Object.create(HTMLElement.prototype);
+
+  proto.createdCallback = function ssb_createdCallback() {
+
+    this._template = template.content.cloneNode(true);
+
+    this.listElem = this._template.getElementById('scrollbox-list');
+    this.createShadowRoot().appendChild(this._template);
+
+    // left and right margin of the scroll list
+    this.margin = this.hasAttribute('margin')?
+                                    this.getAttribute('margin') : 0;
+    this.margin = parseInt(this.margin, 10) * 10;
+
+    this.viewWidth = parseInt(this.clientWidth - 2 * this.margin, 10);
+
+    this.translateX = 0;
+
+    // Current active node
+    this.currentNode = null;
+
+    // add navigable elements to spacial navigator
+    var navigableElems = Array.prototype.slice.call(
+                                  this.getElementsByClassName('navigable'));
+    if (navigableElems.length > 0) {
+      this._scrollTo(navigableElems[0]);
+    }
+    this._spatialNavigator = new SpatialNavigator(navigableElems);
+    this._spatialNavigator.on('focus', this._handleFocus.bind(this));
+    this._spatialNavigator.on('unfocus', this._handleUnfocus.bind(this));
+
+    // handle appendChild and removeChild in scrollbox
+    this._observer = new MutationObserver(this._handleMutation.bind(this));
+    this._observer.observe(this, {
+      childList: true
+    });
+
+    this.addEventListener('transitionend', this);
+    this.addEventListener('scroll', this);
+  };
+
+  /**
+   * Current focused element
+   */
+  Object.defineProperty(proto, 'currentElem', {
+    get: function() {
+      return this._spatialNavigator.getFocusedElement();
+    }
+  });
+
+  /**
+   * First node in scrollbox
+   */
+  Object.defineProperty(proto, 'firstNode', {
```

Comment:
I don't think we need to override `firstNode` and `lastNode`. It should works as the original DOM API. The content may be a little different but it still works. If you want to find the first and last element, you may use `firstElementChild` and `lastElementChild` which are provided by DOM API.